compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUAttrs.cpp

```diff
@@ -1029,6 +1032,78 @@ bool TargetAttr::supportsSyncMMAOps() const {
   return false;
 }
 
+//===----------------------------------------------------------------------===//
+// Lowering Config Attributes
+//===----------------------------------------------------------------------===//
+
+static constexpr char kWorkgroupLevelName[] = "workgroup";
+static constexpr char kPartialReductionLevelName[] = "partial_reduction";
+static constexpr char kReductionLevelName[] = "reduction";
+static constexpr char kThreadLevelName[] = "thread";
+static constexpr char kSubgroupLevelName[] = "subgroup";
+static constexpr char kLaneLevelName[] = "lane";
+
+static StringRef getTilingLevelName(GPU::TilingLevel level) {
+  switch (level) {
+  case GPU::TilingLevel::Workgroup:
+    return kWorkgroupLevelName;
+  case GPU::TilingLevel::PartialReduction:
+    return kPartialReductionLevelName;
+  case GPU::TilingLevel::Reduction:
+    return kReductionLevelName;
+  case GPU::TilingLevel::Thread:
+    return kThreadLevelName;
+  case GPU::TilingLevel::Subgroup:
+    return kSubgroupLevelName;
+  case GPU::TilingLevel::Lane:
+    return kLaneLevelName;
+  }
+  assert(false && "Unknown tiling level");
+  return StringAttr();
+}
+
+static SmallVector<int64_t> getTileSizes(DictionaryAttr config,
+                                         GPU::TilingLevel level) {
+  auto sizes = config.getAs<ArrayAttr>(getTilingLevelName(level));
+  if (!sizes) {
+    return {};
+  }
+  SmallVector<int64_t> sizeList;
+  for (auto maybeSize : sizes.getValue()) {
+    auto size = dyn_cast<IntegerAttr>(maybeSize);
+    if (!size) {
+      return {};
+    }
+    sizeList.push_back(size.getInt());
+  }
```

Comment:
nit: you can do `auto sizeList = to_vector_of<int64_t>(filter_range(sizes.getValue(), IsaPred<IntegerAttr>));`

Reply:
I couldn't get this to work so I did `map_to_vector` instead.